source/common/upstream/upstream_impl.h

```diff
@@ -931,12 +952,16 @@ class ClusterInfoImpl : public ClusterInfo,
   const std::string maintenance_mode_runtime_key_;
   std::shared_ptr<UpstreamLocalAddressSelector> upstream_local_address_selector_;
   LoadBalancerType lb_type_;
-  absl::optional<envoy::config::cluster::v3::Cluster::RoundRobinLbConfig> lb_round_robin_config_;
-  absl::optional<envoy::config::cluster::v3::Cluster::LeastRequestLbConfig>
+  const std::unique_ptr<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>
+      lb_round_robin_config_;
```

Comment:
Are they all optional or is there a requirement that there will be at least one LbPoilcy?
If all are optional, consider having all of them in a single struct to reduce a bit more memory.
This can be done in a future PR.

Reply:
Maybe sometimes there aren't any. LoadBalancerType has 8 values, but there are only 5 configs here. A policy like "Random" is probably config-less.

Reply:
Does only 1 of them usually being set make them a good candidate for `PackedStruct` ? #24743

Reply:
> Does only 1 of them usually being set make them a good candidate for `PackedStruct` ? #24743

Doesn't `PackedStruct` require them to be of the same type?

Reply:
I think this is a discussed for a follow PR!

Reply:
They would have to be stored as uniqe_ptrs to base class. (We can discuss in said follow-up PR)

Reply:
Usually, 0 or 1 of them will be set.

Reply:
`variant` may be a good choice. But not sure if it's valueable to use it just for 20-30 bytes memory saving.

Reply:
20-30 bytes quickly accumulate when you have tens of thousands of clusters. Any savings is valuable!